src/test/fuzz/fuzz.cpp

```diff
@@ -155,8 +158,13 @@ fuzz(std::string const& filename, el::Level logLevel,
     }
     while (loop.getAcceptor()->isConnected())
     {
-        clock.crank(true);
+        testutil::crankSome(clock);
     }
+
+    // manually shutdown everything
+    app1->getWorkScheduler().shutdown();
```

Comment:
You need to crank the clock properly until WorkScheduler is in `BasicWork::State::WORK_ABORTED` state (see `testutil::shutdownWorkScheduler`)

Reply:
The original issue, and why I needed to change any code in the first place is that there are two applications, both using the same clock. Upon executing the current code, at the end of the scope, one app is able to successfully shutdown:

```
 <fuzz> [Work TRACE] Shutting down: work-scheduler [BasicWork.cpp:52]                                                               
 <fuzz> [Work DEBUG] work work-scheduler : WORK_WAITING -> WORK_ABORTING                                                            
 <fuzz> [Work TRACE] work-scheduler: waiting for children to abort. [Work.cpp:92]                                                   
 <fuzz> [Work TRACE] Abort progress for work-scheduler: done [BasicWork.cpp:342]                                                    
 <fuzz> [Work TRACE] resetting work-scheduler [BasicWork.cpp:133]
 <fuzz> [Work DEBUG] work work-scheduler : WORK_ABORTING -> WORK_ABORTED                                                            
 <fuzz> [default DEBUG] Joining 11 worker threads
 <fuzz> [default DEBUG] Joined all 11 threads
 <fuzz> [default INFO] Application destroyed
```

However, the second one just stalls:
```
<fuzz> [Work TRACE] Shutting down: work-scheduler [BasicWork.cpp:52]                                                               
<fuzz> [Work DEBUG] work work-scheduler : WORK_WAITING -> WORK_ABORTING   
```

The two solutions I came up with for this (one Marta and I discussed and one I worked out with the help of Graydon) is to:

1. Have two clocks. Two clocks will not stall on destruction, but based on comments in the code, is not recommended for simulations such as this.

2. Copy the logic from `testutil::shutdownWorkScheduler`. **In my revised code, I have decided to go with this approach**. Basically here, I just manually shutdown the two (not 100% if the stalling app is the same one each time) and then loop&crank until everything is done shutting down.

Reply:
Furthermore, *ideally I would not want to start up and shutdown every single time I inject fuzzed messages.* This will very likely be fixed in the PR following this one -- and removal of the `goto` :)

Reply:
Dealt with by utilizing simulation. See below comments by Nicolas.